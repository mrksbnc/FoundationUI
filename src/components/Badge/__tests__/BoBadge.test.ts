import {
	BadgeSize,
	BadgeType,
	BadgeVariant,
	BoBadge,
} from '@/components/Badge';
import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

describe('BoIcon', () => {
	it('renders properly', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'test',
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders with 'test' label", () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'test',
			},
		});
		expect(wrapper.text()).toBe('test');
	});

	it('renders with correct size', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'test',
				size: BadgeSize.XS,
			},
		});
		expect(wrapper.classes()).toContain('xs');
	});

	it('renders with correct variant', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'test',
				variant: BadgeVariant.Info,
			},
		});
		expect(wrapper.classes()).toContain('info');
	});

	it('renders with correct type', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'test',
				type: BadgeType.Filled,
			},
		});
		expect(wrapper.classes()).toContain('filled');
	});
});
