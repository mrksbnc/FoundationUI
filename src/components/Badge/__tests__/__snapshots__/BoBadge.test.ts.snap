// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoIcon > renders properly 1`] = `
"<div data-v-dfbd0d07="" class="bo-badge default pill filled sm">
  <!--v-if-->
  <!--v-if--><span data-v-dfbd0d07="" class="label">test</span>
  <!--v-if-->
</div>"
`;
